src/binding/python/RecordComponent.cpp

```diff
@@ -352,6 +354,219 @@ store_chunk(RecordComponent & r, py::array & a, py::tuple const & slices)
     store_chunk(r, a, offset, extent, flatten);
 }
 
+struct PythonDynamicMemoryView
+{
+    using ShapeContainer = pybind11::array::ShapeContainer;
+
+    template< typename T >
+    PythonDynamicMemoryView(
+        DynamicMemoryView< T > dynamicView,
+        ShapeContainer arrayShape,
+        ShapeContainer strides )
+        : m_dynamicView( std::shared_ptr< void >(
+              new DynamicMemoryView< T >( std::move( dynamicView ) ) ) )
+        , m_arrayShape( std::move( arrayShape ) )
+        , m_strides( std::move( strides ) )
+        , m_datatype( determineDatatype< T >() )
+    {
+    }
+
+    pybind11::memoryview currentView() const;
+
+    std::shared_ptr< void > m_dynamicView;
+    ShapeContainer m_arrayShape;
+    ShapeContainer m_strides;
+    Datatype m_datatype;
+};
+
+namespace
+{
+struct GetCurrentView
+{
+    template< typename T >
+    pybind11::memoryview
+    operator()( PythonDynamicMemoryView const & dynamicView )
+    {
+        auto span = static_cast< DynamicMemoryView< T > * >(
+            dynamicView.m_dynamicView.get() )->currentBuffer();
+        return py::memoryview::from_buffer(
+            span.data(),
+            dynamicView.m_arrayShape,
+            dynamicView.m_strides,
+            /* readonly = */ false );
+    }
+
+    std::string errorMsg = "DynamicMemoryView";
+};
+
+template<>
+pybind11::memoryview
+GetCurrentView::operator()< std::string >( PythonDynamicMemoryView const & )
+{
+    throw std::runtime_error( "[DynamicMemoryView] Only PODs allowed." );
+}
+} // namespace
+
+pybind11::memoryview PythonDynamicMemoryView::currentView() const
+{
+    static GetCurrentView cv;
```

Comment:
It's an empty struct just used for the switchType template. But thinking about this again, `static` isn't really needed here, but I can make it `const`.
Maybe we should make `switchType` accept its "functor" parameter only as a template parameter instead of an actual function parameter to avoid this? I think there is only one place where this struct is non-empty and that can be easily rewritten.
I.e. something like
```cpp
template< typename Action, typename... Args >
auto switchType( Datatype dt, Args &&... args ) -> […]
```
instead of 
```cpp
template< typename Action, typename... Args >
auto switchType( Datatype dt, Action action, Args &&... args ) -> […]
```

Reply:
I think the current way (with the parameter) is more general since you can always pass in an argument `dt, Action{}, ...` to get the first case.

But if we know that our switch type actions are always stateless then we can also rework it that way.

Reply:
If you need the state, you can always define it as one of the `Args&&...` parameters, so having the `action` as a parameter is a duplication of some sort.

Reply:
True, agreed :)